core/trino-main/src/test/java/io/trino/server/security/TestResourceSecurity.java

```diff
@@ -665,6 +674,98 @@ public HttpCookie getNonceCookie()
         }
     }
 
+    @DataProvider(name = "groups")
+    public static Object[][] groups()
+    {
+        return new Object[][] {
+                {Optional.empty()},
+                {Optional.of(ImmutableSet.of())},
+                {Optional.of(ImmutableSet.of("admin", "public"))}
+        };
+    }
+
+    @Test(dataProvider = "groups")
+    public void testOAuth2Groups(Optional<Set<String>> groups)
+            throws Exception
+    {
+        try (TokenServer tokenServer = new TokenServer(Optional.empty());
+                TestingTrinoServer server = TestingTrinoServer.builder()
+                        .setProperties(ImmutableMap.<String, String>builder()
```

Comment:
The configuration is like giant copy paste, can we somehow extract common configuration and reuse it between tests?